DataMigration/src/test/java/org/sagebionetworks/dian/datamigration/PathsHelperTests.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * BSD 3-Clause License
+ *
+ * Copyright 2021  Sage Bionetworks. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * 1.  Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * 2.  Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation and/or
+ * other materials provided with the distribution.
+ *
+ * 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission. No license is granted to the trademarks of
+ * the copyright holders even if such marks are included in this software.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.sagebionetworks.dian.datamigration;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class PathsHelperTests {
+
+    private final Path resourceDirectory = Paths.get("src", "test", "resources");
+    private final Path rootFileTestFolder = resourceDirectory.resolve("fileTests");
+    private final Path folderA = rootFileTestFolder.resolve("FolderA");
+    private final Path folderB = rootFileTestFolder.resolve("FolderB");
+    private final Path folderUnitTestsDir = resourceDirectory.resolve("folderUnitTests");
+
+    @Before
+    public void before() throws IOException {
+        PathsHelper.deleteDirectoryRecursively(folderUnitTestsDir);
+    }
+
+    @After
+    public void after() throws IOException {
+        PathsHelper.deleteDirectoryRecursively(folderUnitTestsDir);
+    }
```

Comment:
Yes, the unit tests do make temporary filesystem changes, and I tested that even if a unit test fails or crashes, the filesystem is cleaned up in the after() func.  This has been tested both locally and on the github environment.

Java isn't my strongest language, so this may be obvious, but is your comment a conceptual/architectural issue, or is there something different about the way java unit tests are run vs. integration tests?

Reply:
IMO unit tests should mock all external dependencies, including things like the filesystem. They should run entirely in memory, they should test one class at a time—thoroughly. Integration tests verify that the components (like your code and the operating system) work together correctly to fulfill a function.